src/libstd/sys/windows/net.rs

```diff
@@ -96,12 +96,16 @@ impl Socket {
             SocketAddr::V6(..) => c::AF_INET6,
         };
         let socket = unsafe {
-            match c::WSASocketW(fam, ty, 0, ptr::null_mut(), 0,
-                                c::WSA_FLAG_OVERLAPPED) {
+            #[cfg(not(target_os = "uwp"))]
+            let socket_flags = c::WSA_FLAG_OVERLAPPED;
+            #[cfg(target_os = "uwp")]
+            let socket_flags = c::WSA_FLAG_OVERLAPPED | c::WSA_FLAG_NO_HANDLE_INHERIT;
```

Comment:
This might be a good case for `cfg!`

Reply:
(also, is this NO_HANDLE_INHERIT flag available on the normal platform?)

Reply:
Since Windows 7 SP1, and actually used to be in the codebase but [was removed in 2015](https://github.com/rust-lang/rust/pull/26658), interestingly enough.

Reply:
I tried using `cfg!` here, but that would require flagging the `WSA_FLAG_NO_HANDLE_INHERIT` as `allow(dead_code)`. If you prefer this variant I'll do it!

My understanding is that rust targets Windows XP for the resulting applications (and only Windows 7+ for running the compiler itself), is that correct? Dropping XP & Vista would definitely help a lot in sharing more code, but I doubt that's an option :)

Reply:
Down to windows XP? I don't think MSVC itself even supports that anymore, starting at like VS 2017 theres an entirely separate component/target for XP support in VS itself.

I googled it and according to [this](https://forge.rust-lang.org/platform-support.html) Rust targets windows 7+

Reply:
I'll comment later in the thread about our platforms support, but thanks for digging that up @DianaNites! @chouquette if you're willing I think in light of that the best thing to do here would be to actually improve the existing code for Windows by unconditionally passing `WSA_FLAG_HANDLE_NO_INHERIT` and then handling the error mentioned in https://github.com/rust-lang/rust/issues/26543 specifically on non-uwp targets only and falling back to what the code currently does today. Would you be up for configuring it that way?

Reply:
Oh and FWIW it's probably easiest to just slap `#[allow(dead_code)]` in `c.rs`, we don't benefit a whole lot from cleaning that file up between targets.

Reply:
Done in the latest patch, but to be fair I'm not sure it improved the readability (but that might be because my rust skillz are not optimal :D)